vnext/Microsoft.ReactNative/JsiWriter.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "jsi/jsi.h"
+#include "winrt/Microsoft.ReactNative.h"
+
+namespace winrt::Microsoft::ReactNative {
+
+struct JsiWriter : winrt::implements<JsiWriter, IJSValueWriter> {
+  JsiWriter(facebook::jsi::Runtime &runtime) noexcept;
+  facebook::jsi::Value MoveResult() noexcept;
+  facebook::jsi::Value CopyResult() noexcept;
+
+ public: // IJSValueWriter
+  void WriteNull() noexcept;
+  void WriteBoolean(bool value) noexcept;
+  void WriteInt64(int64_t value) noexcept;
+  void WriteDouble(double value) noexcept;
+  void WriteString(const winrt::hstring &value) noexcept;
+  void WriteObjectBegin() noexcept;
+  void WritePropertyName(const winrt::hstring &name) noexcept;
+  void WriteObjectEnd() noexcept;
+  void WriteArrayBegin() noexcept;
+  void WriteArrayEnd() noexcept;
+
+ public:
+  static facebook::jsi::Value ToJsiValue(facebook::jsi::Runtime &runtime, JSValueArgWriter const &argWriter) noexcept;
+
+ private:
+  enum class ContinuationAction {
+    AcceptValueAndFinish,
+    AcceptArrayElement,
+    AcceptPropertyName,
+    AcceptPropertyValue,
+  };
+
+  struct Continuation {
```

Comment:
>Continuation [](start = 9, length = 12)

Just wonder about the term: why is it a Continuation?

Reply:
I didn't come out with a good idea. I call it a continuation because, it defines what to do after receiving a value.

Reply:
I personally like the Container term used in the JSValueTreeWriter.h. It makes code smaller and easier to read. We essentially build a stack of containers which are either Object or Array.

---
In reply to: [430718989](https://github.com/microsoft/react-native-windows/pull/5009#discussion_r430718989) [](ancestors = 430718989)

Reply:
I will reconsider it